config.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2020 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+package opa
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+)
+
+const wasmPageSize = 65535
+
+type config struct {
+	Policy         []byte
+	Data           []byte
+	MemoryMinPages uint32
+	MemoryMaxPages uint32 // 0 means no limit.
+	PoolSize       uint32
+	ErrorLogger    func(err error)
+}
+
+// PolicyFile configures a policy file to load.
+func PolicyFile(fileName string) func(c *config) error {
```

Comment:
Nit: I don't have a strong feeling about this but I've found this functional argument pattern to be annoying when you want to construct objects dynamically (you end up having to build a slice and varargs.) I've found the builder pattern is a bit more future proof. E.g., `err := config.WithPolicyFile(xxx).WithDataFile(yyy).DoSomething()`

Reply:
Sure. Replaced with With() construct, followed by Init() to get the errors out.